CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+# Contributing to Caliptra
+
+## Contributor License Agreement
+
+Per the [Charter](Caliptra_WG_Technical_Charter.md), contributors to Caliptra
+must have signed the CHIPS CLA.
+
+## Guidelines for code contributions
+
+* Changes must be made via pull request
+* [Branch protection rules](github_rules.md) ensure pull requests are reviewed
+  by project members
+* [Maintainers](MAINTAINERS.md) are expected to enforce `CODEOWNERS` rights
+  where necessary. Maintainers are also the sole owners of the repo's
+  configuration.
+* The branch protection rules include constraints to maintain a linear history
+  and avoid merges in pull requests.
+* Please keep a clean commit history, and no long series
+  of "fixup" patches. Use GitHub's "Squash and merge" option if necessary.
+* New code files must include the Apache license 2.0 header. Commit verification
```

Comment:
We'll need to revisit this policy once we start vendoring third-party firmware deps. 

I expect we'll want something like this:

"With the exception of the Rust toolchain, all third-party code necessary for building release firmware binaries must be placed in the caliptra-sw/third-party directory."

Reply:
Applied a bit more

+* With the exception of the Rust toolchain, all third-party code necessary for
+  building release firmware binaries must be placed in the
+  `caliptra-sw/third-party` directory
+* Third party vendored repos included into Caliptra repos may not be ASL 2.0
+  license. However, all vendored repos must be ASL 2.0 compliant and not
+  introduce additional restrictions or encumbrances. Ask the TAC if in doubt.